### DISCUSSION TITLE
Accessing the steps counter in an agent function

### DISCUSSION THREAD

**Discussion by isolin** [FLAMEGPU/FLAMEGPU2#775] [Q&A] [answered]
I would like to process the `tick` in my agent's function. According to the docs this should be possible through `CUDASimulation.getStepCounter()` but I hav eno idea how to access it from a `FLAMEGPU_AGENT_FUNCTION`. Of course the following fails:
```cuda
FLAMEGPU_AGENT_FUNCTION(agent_update, flamegpu::MessageNone, flamegpu::MessageBruteForce) {
    //...
    const unsigned int timestep = FLAMEGPU->CUDASimulation.getStepCounter();
    //...
}
```

So how can I get the simulation timestep other than storing and incrementing it explicitly in the agent as one of its variables (that would be a serious overhead as the single value would unnecessarily replicate throughout the whole population)?

**Comment by Robadob** [comment, marked as answer]
I think the method you are looking for is `FLAMEGPU->getStepCounter();`

https://github.com/FLAMEGPU/FLAMEGPU2/blob/cf64fd9230819f4cc1a092202c3b94a7efbbad8e/include/flamegpu/runtime/DeviceAPI.cuh#L99

You should be able to access it the same in both Host and Device APIs.
